pkg/storage/cacher.go

```diff
@@ -463,7 +463,21 @@ func (c *Cacher) List(ctx context.Context, key string, resourceVersion string, p
 
 // Implements storage.Interface.
 func (c *Cacher) GuaranteedUpdate(ctx context.Context, key string, ptrToType runtime.Object, ignoreNotFound bool, preconditions *Preconditions, tryUpdate UpdateFunc) error {
-	return c.storage.GuaranteedUpdate(ctx, key, ptrToType, ignoreNotFound, preconditions, tryUpdate)
+	var currObj runtime.Object
+	if elem, exists, err := c.watchCache.GetByKey(key); err == nil && exists {
```

Comment:
This is POC - sure, we shouldn't ignore it. Just wanted to get bu-in from you before cleaning it.